### PR TITLE
Dockerfile: Add torsocks binary.

### PR DESCRIPTION
**Pull request by schildbach**
This has two purposes:
1) The 'torify' wrapper is already included, and it needs torsocks.
2) Use torsocks for health checking your onion service like this:
``` 
healthcheck:
  test: ["CMD-SHELL", "torsocks nc -z <onion address> <port> || exit 1"]
  start_period: 2m
  interval: 1m
```

This is a draft, though I'm already running this in production.

This PR currently includes #54, I'll rebase once that is merged.

I've hardcoded this to torsocks v2.3.0 atm because I don't know what works best for your project. We could also use the latest master, or make the tag configurable via build argument. Or we could install torsocks from the Alpine repository, but I assume it would also pull in a second copy of tor.

Also, actually I don't know if it is a good idea from a privacy point of view to connect to a hidden service that's on the machine.

### CODE REVIEW

**Comment by cmehay** [comment]
I don't mind having torsock binary in the container. 

Concerning the version, maybe should we use the script logic used to get the last tor version ?

https://github.com/cmehay/docker-tor-hidden-service/blob/c08f702da12c89160d016b690e556c76fac0e276/last_tor_version.sh#L2

**Comment by schildbach** [comment]
I just pushed an update to my PR that adds the last_torsocks_version.sh logic and uses it in the build.

File: Dockerfile [hunk 1]
```diff
@@ -17,8 +18,17 @@ RUN     apk add --no-cache git bind-tools libevent-dev openssl-dev gnupg gcc mak
...
     rm -rf tor && \
+    git clone https://git.torproject.org/torsocks.git /usr/local/src/torsocks && \
+    cd /usr/local/src/torsocks && \
+    git checkout $torsocks_version && \
+    apk add --no-cache libtool && \
+    ./autogen.sh && \
+    ./configure && \
+    make && make install && \
+    cd .. && \
+    rm -rf torsocks && \
```
Comment (cmehay) [inline Dockerfile:29]: I think it should be better to add another RUN instruction for torsock for a better readability, it won't be a issue for image size as long we remove every installed packages on each RUN instructions.

**Review by cmehay — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (schildbach) [reply Dockerfile:29]: Regarding image size, I plan another PR that uses a fresh image for runtime. Currently it inherits its context from the build image and the current directory of the docker invocation. The way it is currently done pulls in tons and tons of MB of useless (and potentially dangerous) binaries and data.

Comment (schildbach) [reply Dockerfile]: Ok, I extracted my changes into a separate, self-contained RUN instruction.

**Review by cmehay — APPROVED** [review]

**Pull request merged**